docs/ADR-X-Better-call-stacks-of-io-exceptions.md

```diff
@@ -0,0 +1,79 @@
+# Status
+
+📜 Proposed 2025-03-04
+
+# Context
+
+This ADR enriches exceptions thrown from `IO` functions with call stacks up to the function usage site.
+
+## The problem
+This ADR bases on [[ADR-8-Use-RIO-in-cardano‐cli]].
+
+The goal of this ADR is to provide better call stacks in `IO` exceptions.
+The problem with exceptions thrown from IO monad is that they do not carry call stack from where they were thrown.
+This is not very convenient when for example doing multiple file access or network operations in a row.
+You won't know which one has thrown an exception: you will have only a vague exception message without many details, for example:
+```
+Network.Socket.recvBuf: resource vanished (Connection reset by peer)
+```
+
+## Solution
+
+The currently used type alias adds call stacks to the functions:
+```haskell
+type CIO e a = HasCallStack => CIO e a
```

Comment:
ADR 8 proposes to [use the RIO monad as the return type](https://github.com/input-output-hk/cardano-node-wiki/blob/main/docs/ADR-8-Use-RIO-in-cardano%E2%80%90cli.md#proposed-solution) in many places. How will `CIO` and `RIO` work together?

Reply:
oh that's a typo. It should be:
```haskell
type CIO e a = HasCallStack => RIO e a
```